NullPointerException when accessing node details

Running both server and web v1.1.4 in Ubuntu (ElasticSearch v1.6.0), I get the following NullPointerException when I click the only node's details on any part of the web interface. Everything else works flawlessly, even heap usage is displayed correctly and the metrics page is accessible.

```
2015-07-07T13:42:53.747Z - [ERROR] - from play in play-akka.actor.default-dispatcher-3 
Cannot invoke the action, eventually got an error: java.lang.NullPointerException

2015-07-07T13:42:53.751Z - [ERROR] - from application in play-akka.actor.default-dispatcher-3 


! @6mlai55kh - Internal server error, for (GET) [/system/nodes/ebb19cce-4162-4e49-822f-3ed6c3591d5e] ->

play.api.Application$$anon$1: Execution exception[[NullPointerException: null]]
	at play.api.Application$class.handleError(Application.scala:296) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.api.DefaultApplication.handleError(Application.scala:402) [com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3$$anonfun$applyOrElse$4.apply(PlayDefaultUpstreamHandler.scala:320) [com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3$$anonfun$applyOrElse$4.apply(PlayDefaultUpstreamHandler.scala:320) [com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at scala.Option.map(Option.scala:145) [org.scala-lang.scala-library-2.10.4.jar:na]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3.applyOrElse(PlayDefaultUpstreamHandler.scala:320) [com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3.applyOrElse(PlayDefaultUpstreamHandler.scala:316) [com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:344) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:343) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32) [org.scala-lang.scala-library-2.10.4.jar:na]
	at play.api.libs.iteratee.Execution$trampoline$.execute(Execution.scala:46) [com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:40) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:248) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.Promise$class.complete(Promise.scala:55) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:153) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32) [org.scala-lang.scala-library-2.10.4.jar:na]
	at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.processBatch$1(BatchingExecutor.scala:67) [com.typesafe.akka.akka-actor_2.10-2.3.5.jar:na]
	at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.apply$mcV$sp(BatchingExecutor.scala:82) [com.typesafe.akka.akka-actor_2.10-2.3.5.jar:na]
	at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.apply(BatchingExecutor.scala:59) [com.typesafe.akka.akka-actor_2.10-2.3.5.jar:na]
	at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.apply(BatchingExecutor.scala:59) [com.typesafe.akka.akka-actor_2.10-2.3.5.jar:na]
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:72) [org.scala-lang.scala-library-2.10.4.jar:na]
	at akka.dispatch.BatchingExecutor$Batch.run(BatchingExecutor.scala:58) [com.typesafe.akka.akka-actor_2.10-2.3.5.jar:na]
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:41) [com.typesafe.akka.akka-actor_2.10-2.3.5.jar:na]
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:393) [com.typesafe.akka.akka-actor_2.10-2.3.5.jar:na]
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979) [org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107) [org.scala-lang.scala-library-2.10.4.jar:na]
Caused by: java.lang.NullPointerException: null
	at views.html.system.nodes.show$.apply(show.template.scala:175) ~[graylog-web-interface.graylog-web-interface-1.1.4.jar:1.1.4]
	at views.html.system.nodes.show$.render(show.template.scala:306) ~[graylog-web-interface.graylog-web-interface-1.1.4.jar:1.1.4]
	at views.html.system.nodes.show.render(show.template.scala) ~[graylog-web-interface.graylog-web-interface-1.1.4.jar:1.1.4]
	at controllers.NodesController.node(NodesController.java:106) ~[graylog-web-interface.graylog-web-interface-1.1.4.jar:1.1.4]
	at Routes$$anonfun$routes$1$$anonfun$applyOrElse$43$$anonfun$apply$567.apply(routes_routing.scala:1947) ~[graylog-web-interface.graylog-web-interface-1.1.4.jar:na]
	at Routes$$anonfun$routes$1$$anonfun$applyOrElse$43$$anonfun$apply$567.apply(routes_routing.scala:1947) ~[graylog-web-interface.graylog-web-interface-1.1.4.jar:na]
	at play.core.Router$HandlerInvokerFactory$$anon$4.resultCall(Router.scala:264) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.core.Router$HandlerInvokerFactory$JavaActionInvokerFactory$$anon$15$$anon$1.invocation(Router.scala:255) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.core.j.JavaAction$$anon$1.call(JavaAction.scala:55) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.GlobalSettings$1.call(GlobalSettings.java:67) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.mvc.Security$AuthenticatedAction.call(Security.java:44) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.core.j.JavaAction$$anonfun$11.apply(JavaAction.scala:82) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.core.j.JavaAction$$anonfun$11.apply(JavaAction.scala:82) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24) ~[org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24) ~[org.scala-lang.scala-library-2.10.4.jar:na]
	at play.core.j.HttpExecutionContext$$anon$2.run(HttpExecutionContext.scala:40) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.api.libs.iteratee.Execution$trampoline$.execute(Execution.scala:46) [com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]
	at play.core.j.HttpExecutionContext.execute(HttpExecutionContext.scala:32) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at scala.concurrent.impl.Future$.apply(Future.scala:31) ~[org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.Future$.apply(Future.scala:485) ~[org.scala-lang.scala-library-2.10.4.jar:na]
	at play.core.j.JavaAction$class.apply(JavaAction.scala:82) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.core.Router$HandlerInvokerFactory$JavaActionInvokerFactory$$anon$15$$anon$1.apply(Router.scala:252) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4$$anonfun$apply$5.apply(Action.scala:130) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4$$anonfun$apply$5.apply(Action.scala:130) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.utils.Threads$.withContextClassLoader(Threads.scala:21) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4.apply(Action.scala:129) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4.apply(Action.scala:128) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at scala.Option.map(Option.scala:145) [org.scala-lang.scala-library-2.10.4.jar:na]
	at play.api.mvc.Action$$anonfun$apply$1.apply(Action.scala:128) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.api.mvc.Action$$anonfun$apply$1.apply(Action.scala:121) ~[com.typesafe.play.play_2.10-2.3.9.jar:2.3.9]
	at play.api.libs.iteratee.Iteratee$$anonfun$mapM$1.apply(Iteratee.scala:483) ~[com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]
	at play.api.libs.iteratee.Iteratee$$anonfun$mapM$1.apply(Iteratee.scala:483) ~[com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]
	at play.api.libs.iteratee.Iteratee$$anonfun$flatMapM$1.apply(Iteratee.scala:519) ~[com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]
	at play.api.libs.iteratee.Iteratee$$anonfun$flatMapM$1.apply(Iteratee.scala:519) ~[com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]
	at play.api.libs.iteratee.Iteratee$$anonfun$flatMap$1$$anonfun$apply$14.apply(Iteratee.scala:496) ~[com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]
	at play.api.libs.iteratee.Iteratee$$anonfun$flatMap$1$$anonfun$apply$14.apply(Iteratee.scala:496) ~[com.typesafe.play.play-iteratees_2.10-2.3.9.jar:2.3.9]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24) ~[org.scala-lang.scala-library-2.10.4.jar:na]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24) ~[org.scala-lang.scala-library-2.10.4.jar:na]
	... 6 common frames omitted
```

--- a/app/views/system/nodes/show.scala.html
+++ b/app/views/system/nodes/show.scala.html
@@ -1,12 +1,10 @@
 @(currentUser: org.graylog2.restclient.models.User, breadcrumbs: lib.BreadcrumbList, node: org.graylog2.restclient.models.Node, installedPlugins: List[org.graylog2.restclient.models.Plugin])
-@import views.helpers.StringHelper._
-@import views.helpers.DateHelper
-@import controllers.routes;
-@import views.helpers.Permissions.isPermitted
+@import controllers.routes
 @import lib.security.RestPermissions._
-@import org.joda.time.DateTime
-@import org.joda.time.DateTimeZone
-@import views.helpers.DocsHelper
+@import org.joda.time.{DateTime, DateTimeZone}
+@import views.helpers.Permissions.isPermitted
+@import views.helpers.StringHelper._
+@import views.helpers.{DateHelper, DocsHelper}
 
 
     @main("Node " + node.getNodeId, null, "", currentUser, false) {
@@ -141,19 +139,24 @@
         </div>
     </div>
 
-
-
     <div class="row content">
         <div class="col-md-12">
+        @if(node.getJournalInfo.enabled) {
             <div id="react-journal-info"
-                data-node-id="@node.getNodeId"
-                data-journal-enabled="@node.getJournalInfo.enabled"
-                data-journal-dir="@node.getJournalInfo.journalConfig.directory"
-                data-journal-max-size="@node.getJournalInfo.journalConfig.maxSize"
-                data-journal-maxage="@node.getJournalInfo.journalConfig.maxAge"
-                data-journal-flush-interval="@node.getJournalInfo.journalConfig.flushInterval"
-                data-journal-flush-age="@node.getJournalInfo.journalConfig.flushAge">
+            data-node-id="@node.getNodeId"
+            data-journal-enabled="@node.getJournalInfo.enabled"
+            data-journal-dir="@node.getJournalInfo.journalConfig.directory"
+            data-journal-max-size="@node.getJournalInfo.journalConfig.maxSize"
+            data-journal-maxage="@node.getJournalInfo.journalConfig.maxAge"
+            data-journal-flush-interval="@node.getJournalInfo.journalConfig.flushInterval"
+            data-journal-flush-age="@node.getJournalInfo.journalConfig.flushAge">
             </div>
+        } else {
+            <div id="react-journal-info"
+            data-node-id="@node.getNodeId"
+            data-journal-enabled="@node.getJournalInfo.enabled">
+            </div>
+        }
         </div>
     </div>
 

--- a/javascript/src/components/node/JournalDetails.jsx
+++ b/javascript/src/components/node/JournalDetails.jsx
@@ -74,7 +74,7 @@ var JournalDetails = React.createClass({
 
     render() {
         var content = null;
-        if (this.props.enabled) {
+        if (this.props.enabled === "true") {
             var oldestSegment = moment(this.state.oldestSegment);
             var overcommittedWarning = null;
             if (this.state.initialized && (this.state.utilizationRatio >= 1)) {

